idb/postgres/internal/encoding/encode.go

```diff
@@ -53,7 +55,30 @@ func convertEvalDelta(evalDelta types.EvalDelta) types.EvalDelta {
 	return evalDelta
 }
 
+// SanitizeNullForQuery converts a string for putting into a SELECT ... WHERE
+func SanitizeNullForQuery(str string) string {
+	return EscapeNulls(str) // TODO: does Postgres query need double-escaping?
```

Comment:
Alas, the answer is yes. One of the failures tests exactly this case:
```
--- FAIL: TestEmbeddedNullString (87.44s)
    postgres_integration_test.go:76: 
        	Error Trace:	postgres_integration_test.go:76
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestEmbeddedNullString
```

Maybe we need to triple escape now?